docs/100_contribute_docs.md

```diff
@@ -1,4 +1,4 @@
-# About This Documentation
+# Contributing Documentation
```

Comment:
well, "Documentation" term also includes https://dandi.readthedocs.io/ and possibly others. 
So if to rename, I would have said something like

```suggestion
# Contributing to this Handbook
```

but also didn't place it under "Developers ..." section since it is not per se about development -- users and alike are welcome to contribute too...

so overall not sure about this PR

Reply:
@kabilar ping on this PR -- let's finish it up?